sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewViewModel.groovy

```diff
@@ -14,23 +15,25 @@ import life.qbic.datamodel.samples.Sample
 */
 class ProjectOverviewViewModel {
 
-    List<Project> projects = []
+    ObservableList projects = new ArrayList<Project>()
+    final ProjectResourceService projectResourceService
 
-    ProjectOverviewViewModel(){
-        generateMockData()
-    }
-
-    void generateMockData(){
-        Location location = new Location()
+    Subscriber<life.qbic.datamodel.dtos.projectmanagement.Project> projectAddedSubscription
+    Subscriber<life.qbic.datamodel.dtos.projectmanagement.Project> projectRemovedSubscription
 
-        Sample sample = new Sample()
-        sample.setCode("Q1234AC")
-        sample.setCurrentLocation(location)
+    ProjectOverviewViewModel(ProjectResourceService projectResourceService){
+        this.projectResourceService = projectResourceService
+        fetchProjectData()
+        subscribeToResources()
+    }
 
-        Project project1 = new Project.Builder("Q1234","This is a test project",[sample]).progress(0).build()
-        Project project2 = new Project.Builder("Q1244","This is a test project 2",[sample]).progress(1).build()
-        Project project3 = new Project.Builder("Q1233","This is a test project 3",[sample]).progress(0.9).failedSamples(1).build()
+    private void fetchProjectData() {
+        projects.clear()
+        projects.addAll(projectResourceService.iterator())
+    }
 
-        projects.addAll(project1,project2,project3)
+    private void subscribeToResources() {
+        this.projectResourceService.subscribe(projectAddedSubscription, Topic.PROJECT_ADDED)
+        this.projectResourceService.subscribe(projectRemovedSubscription, Topic.PROJECT_REMOVED)
```

Comment:
the added and removed subscriptions are null, they are not initialized yet.

Reply:
I think this is an important catch